smartlearner/views.py

```diff
@@ -51,3 +51,39 @@ def mean(self):
     @property
     def stderror(self):
         return ItemGetter(self, attribute=1)
+
+
+class ReconstructionError(View):
+    def __init__(self, predict_fct, dataset):
+        super().__init__()
+
+        batch_size = 1024  # Internal buffer
+        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
+
+        input = dataset.symb_inputs
+        target = dataset.symb_targets
+        reconstruction_errors = T.sqr(predict_fct(input) - target).mean()
```

Comment:
Of course, if you don't care about the stderr, you don't have to keep the reconstruction error of every examples. You can just take the mean of the average reconstruction error of every batches.

However, if you do want the stderr of your model predictions on the dataset, you'd need to keep the reconstruction error of every examples then compute the mean and the stderr.